warplib/src/warp_memory/accessors.cairo

```diff
@@ -0,0 +1,118 @@
+    fn store<T, impl TSerde: Serde<T>, impl TDrop: Drop<T>>(ref self: WarpMemory, position: felt252, value: T);
+    fn retrieve<T, impl TSerde: Serde<T>, impl TDrop: Drop<T>>(ref self: WarpMemory, position: felt252, size: felt252) -> T;
+    fn create<T, impl TSerde: Serde<T>, impl TDrop: Drop<T>>(ref self: WarpMemory, value: T);
+}
+
+trait WarpMemoryMultiCellAccessorTrait {
+    fn write_multiple(ref self: WarpMemory, position: felt252, ref value: Array::<felt252>);
+    fn read_multiple(ref self: WarpMemory, position: felt252, size: felt252) -> Array::<felt252>;
+}
+
+
+impl WarpMemoryMultiCellAccessor of WarpMemoryMultiCellAccessorTrait {
+    fn write_multiple(ref self: WarpMemory, mut position: felt252, ref value: Array::<felt252>) {
+        let position_256 = u256_from_felt252(position);
+        let len_256: u256 = u256_from_felt252(u32_to_felt252(value.len()));
+        let final_location_256 = position_256 + len_256;
+
+        let MAX_FELT_256 = u256_from_felt252(-1);
+        if final_location_256 > MAX_FELT_256 {
+            panic_with_felt252('Multiple writing overflow');
+        }
+
+        let pointer_256 = u256_from_felt252(self.free_space_pointer);
+        if final_location_256 > pointer_256 {
+            panic_with_felt252('MWriting on unreserved position')
+        }
+
+        loop {
+            match gas::withdraw_gas() {
+               Option::Some(_) => {},
+               Option::None(_) => {
+                    panic_with_felt252('Out of gas');
+               }
+            }
+
+            if value.len() == 0 {
+                break();
+            }
+            self.unsafe_write(position, value.pop_front().unwrap());
+            position += 1;
+        }
+    }
+
+    fn read_multiple(ref self: WarpMemory, position: felt252, size: felt252) -> Array::<felt252> {
+        let start_256 = u256_from_felt252(position);
+        let size_256 = u256_from_felt252(size);
+        let final_location_256 = start_256 + size_256;
+
+        let MAX_FELT_256 = u256_from_felt252(-1);
+        if final_location_256 > MAX_FELT_256 {
+            panic_with_felt252('Multiple reading overflow');
+        }
+
+        let pointer_256 = u256_from_felt252(self.free_space_pointer);
+        if final_location_256 > pointer_256 {
+            panic_with_felt252('MReading on unreserved position')
+        }
+
+        let mut index = position;
+        let mut array: Array<felt252> = ArrayImpl::<felt252>::new();
+        let final_location = position + size;
+        loop {
+            match gas::withdraw_gas() {
+               Option::Some(_) => {},
+               Option::None(_) => {
+                    panic_with_felt252('Out of gas');
+               }
+            }
+
+            if index == final_location {
+                break();
+            }
+            array.append(self.read(index));
+            index += 1;
+        };
+        array
+    }
+}
+
+
+impl WarpMemoryAccesssor of WarpMemoryAccesssorTrait {
```

Comment:
can we write this `WarpMemoryAccesssor ` -> `WarpMemoryAccesssorImpl` as rather , not matter gonna anyway but let's keep this as a naming convention for traits, wdyt ?